src/events/mod.rs

```diff
@@ -833,6 +846,49 @@ impl<'a> arbitrary::Arbitrary<'a> for BytesCData<'a> {
     }
 }
 
+/// Iterator over `CDATA` sections in a string.
+///
+/// This iterator is created by the [`BytesCData::escaped`] method.
+///
+/// If a string contains `]]>`, it needs to be split into multiple `CDATA`
+/// sections, splitting the `]]` and `>` characters. This iterator yields
+/// each of those sections.
+pub struct CDataIterator<'a> {
+    content: &'a str,
+    iter: MatchIndices<'a, &'a str>,
+    start: usize,
+    finished: bool,
```

Comment:
Do we need a separate `finished` field? Isn't enough to rely on `MatchIndices` iterator returning `None`? [`MatchIndices` is `FusedIterator`](https://doc.rust-lang.org/std/str/struct.MatchIndices.html#impl-FusedIterator-for-MatchIndices%3C'a,+P%3E)

Also, the iterator always continues to yield `None` when exhausted, so implement `FusedIterator` for it.

Reply:
> Do we need a separate `finished` field? Isn't enough to rely on `MatchIndices` iterator returning `None`?

AFAICT that is not enough, but I might be wrong. the problem is that the `MatchIndices` gives us the positions of the N splitters (`]]>`), but we need to yield the N+1 parts around the splitters. it might be possible to use a different base iterator that already yields the N+1 relevant parts, but I haven't found one yet.